### PR TITLE
[WIP] change from golide to dep

### PR DESCRIPTION
**Pull request by zhexuany**
Before:
If we run `find _vendor/ -type f | wc -l`, the output is 1108. 
After:
If you run `find vendor/ -type f | wc -l` and the output is 1783. 

I am not sure is it worth to change it or we wait until `dep` is more stable.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/pd/879)

### CODE REVIEW

**Comment by zhexuany** [comment]
Working on it.

File: Makefile [hunk 1]
```diff
@@ -25,18 +33,21 @@ all: dev
 dev: build simulator check test
 
 build:
+	go get -u github.com/golang/dep/cmd/dep
+	dep ensure
+	dep prune
```
Comment (disksing) [inline Makefile]: I think it's better to move them to 'make update'.

Comment (siddontang) [reply Makefile]: +1

**Comment by zhexuany** [comment]
closing for now. This PR is already ruined.

**Pull request closed without merging**